src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -466,13 +466,13 @@ def _mds_rm(self, name):
 
     @orchestrator._cli_write_command(
         'orchestrator rgw add',
-        'name=realm_name,type=CephString '
-        'name=zone_name,type=CephString '
+        'name=realm_name,type=CephString,req=false '
+        'name=zone_name,type=CephString,req=false '
```

Comment:
Last week, I stumbled across this, too. Especially as it is non-obvious that you should probably call this with `default` `default`. :+1: from my side here, except that we'Re now creating an imbalance between `add` and `rm`, as `rm` still requires them.

What about just marking `realm` as `req=false`?

Reply:
but then your still sending in one 'default' and we're suppose to be fully defining the RGW in json. So it's redundant. I agree the imbalance between `add` and `update` or 'rm', did make me question this. But then figured if only add supports `inbuf` or `(-i)` then that's why `req=false` is required.

Of course we could change the useage to:

```
usage = """
Usage:
  ceph orchestrator rgw add <zone_name> -i <json_file>
  ceph orchestrator rgw add <realm_name> <zone_name>
    """
```
But that just doesn't look right.